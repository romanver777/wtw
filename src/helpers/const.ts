export const genres: Array<string> = [
  "Все жанры",
  "Комедии",
  "Документальное",
  "Драма",
  "Ужасы",
  "Семейное",
  "Романтические",
  "Фантастика",
  "Триллеры",
];

export const tabsName: Array<string> = ["Описание", "Информация", "Отзывы"];

export const similarMovies = "Вам может понравиться";

export const page: { [key: string]: string } = {
  movieCard: "movie-card",
  userPage: "user-page",
};

export const userPageTitle = {
  signIn: "Выполните вход",
  myList: "Мой список",
};
