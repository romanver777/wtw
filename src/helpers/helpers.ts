export const convertRatingLevel = (rt: string | null): string => {
  const rating = Number(rt);
  if (rating > 0 && rating < 4) return "Плохо";
  if (rating >= 4 && rating < 5) return "Удовлетворительно";
  if (rating >= 5 && rating < 6) return "Хорошо";
  if (rating >= 6 && rating < 7) return "Очень хорошо";
  if (rating >= 7 && rating < 8) return "Отлично";
  if (rating >= 8) return "Превосходно";
  return "-";
};

export const convertDateTime = (date: string): string => {
  return new Date(date).toLocaleString("ru-RU", {
    year: "numeric",
    month: "long",
    day: "numeric",
    hour: "numeric",
    minute: "numeric",
  });
};
