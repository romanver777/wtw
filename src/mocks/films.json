{
  "films": [
    {
      "filmId": 1302121,
      "nameRu": "Соединённые Штаты против Билли Холидей",
      "nameEn": "The United States vs. Billie Holiday",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "биография"
        },
        {
          "genre": "музыка"
        }
      ],
      "rating": "5.8",
      "ratingVoteCount": 586,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1302121.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1302121.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1171895,
      "nameRu": "Неуязвимый (сериал)",
      "nameEn": "Invincible",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "ужасы"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 1549,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1171895.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1171895.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245486,
      "nameRu": "Невероятные (сериал)",
      "nameEn": "The Nevers",
      "year": "2021-...",
      "filmLength": null,
      "countries": [
        {
          "country": "Великобритания"
        },
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 882,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245486.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245486.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1015471,
      "nameRu": "Райя и последний дракон",
      "nameEn": "Raya and the Last Dragon",
      "year": "2021",
      "filmLength": "1:47",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "мультфильм"
        },
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "7.6",
      "ratingVoteCount": 43386,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1015471.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1015471.jpg",
      "ratingChange": null
    },
    {
      "filmId": 571335,
      "nameRu": "Бесстыжие (сериал)",
      "nameEn": "Shameless",
      "year": "2011-2021",
      "filmLength": "0:55",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "8.6",
      "ratingVoteCount": 207954,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/571335.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/571335.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1328036,
      "nameRu": "Happy End (сериал)",
      "nameEn": null,
      "year": "2021-...",
      "filmLength": "0:35",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "драма"
        },
        {
          "genre": "комедия"
        }
      ],
      "rating": "7.4",
      "ratingVoteCount": 3282,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1328036.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1328036.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1045583,
      "nameRu": "Конёк-Горбунок",
      "nameEn": null,
      "year": "2021",
      "filmLength": "1:51",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "6.8",
      "ratingVoteCount": 100901,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1045583.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1045583.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1274271,
      "nameRu": "Родные",
      "nameEn": null,
      "year": "2020",
      "filmLength": "1:40",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.4",
      "ratingVoteCount": 20539,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1274271.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1274271.jpg",
      "ratingChange": null
    },
    {
      "filmId": 767379,
      "nameRu": "Фарго (сериал)",
      "nameEn": "Fargo",
      "year": "2014-...",
      "filmLength": "0:53",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.4",
      "ratingVoteCount": 119492,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/767379.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/767379.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1245524,
      "nameRu": "Зависнуть в Палм-Спрингс",
      "nameEn": "Palm Springs",
      "year": "2020",
      "filmLength": "1:30",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Гонконг"
        }
      ],
      "genres": [
        {
          "genre": "комедия"
        },
        {
          "genre": "мелодрама"
        },
        {
          "genre": "фантастика"
        }
      ],
      "rating": "7.2",
      "ratingVoteCount": 130542,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1245524.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1245524.jpg",
      "ratingChange": null
    },
    {
      "filmId": 4430018,
      "nameRu": "Абсолютное доказательство",
      "nameEn": "Absolute Proof",
      "year": "2021",
      "filmLength": "2:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "документальный"
        }
      ],
      "rating": null,
      "ratingVoteCount": null,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/4430018.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/4430018.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1437860,
      "nameRu": "Чикатило (сериал)",
      "nameEn": null,
      "year": "2020-2020",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "детектив"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "5.9",
      "ratingVoteCount": 7820,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1437860.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1437860.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1005008,
      "nameRu": "Майор Гром",
      "nameEn": null,
      "year": "2017",
      "filmLength": "0:25",
      "countries": [
        {
          "country": "Россия"
        }
      ],
      "genres": [
        {
          "genre": "короткометражка"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "7.0",
      "ratingVoteCount": 29251,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1005008.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1005008.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1291310,
      "nameRu": "В её глазах (мини-сериал)",
      "nameEn": "Behind Her Eyes",
      "year": "2021-2021",
      "filmLength": "0:50",
      "countries": [
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "детектив"
        }
      ],
      "rating": "7.3",
      "ratingVoteCount": 12949,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1291310.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1291310.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1038789,
      "nameRu": "Охотник на монстров",
      "nameEn": "Monster Hunter",
      "year": "2020",
      "filmLength": "1:43",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Китай"
        },
        {
          "country": "Германия"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        },
        {
          "genre": "приключения"
        }
      ],
      "rating": "5.6",
      "ratingVoteCount": 24933,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1038789.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1038789.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1248786,
      "nameRu": "Аферистка",
      "nameEn": "I Care a Lot",
      "year": "2020",
      "filmLength": "1:58",
      "countries": [
        {
          "country": "США"
        },
        {
          "country": "Великобритания"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "комедия"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "6.6",
      "ratingVoteCount": 81569,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1248786.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1248786.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1193013,
      "nameRu": "Грань времени",
      "nameEn": "Synchronic",
      "year": "2019",
      "filmLength": "1:42",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "драма"
        }
      ],
      "rating": "6.2",
      "ratingVoteCount": 10361,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1193013.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1193013.jpg",
      "ratingChange": null
    },
    {
      "filmId": 958500,
      "nameRu": "Охотник за разумом (сериал)",
      "nameEn": "Mindhunter",
      "year": "2017-2019",
      "filmLength": "1:00",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "триллер"
        },
        {
          "genre": "драма"
        },
        {
          "genre": "криминал"
        }
      ],
      "rating": "8.1",
      "ratingVoteCount": 50186,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/958500.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/958500.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1347194,
      "nameRu": "Звездный рубеж",
      "nameEn": "Cosmic Sin",
      "year": "2021",
      "filmLength": "1:28",
      "countries": [
        {
          "country": "США"
        }
      ],
      "genres": [
        {
          "genre": "фантастика"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "3.4",
      "ratingVoteCount": 524,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1347194.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1347194.jpg",
      "ratingChange": null
    },
    {
      "filmId": 1236082,
      "nameRu": "Ассасин: Битва миров",
      "nameEn": "Ci sha xiao shuo jia",
      "year": "2021",
      "filmLength": "2:10",
      "countries": [
        {
          "country": "Китай"
        }
      ],
      "genres": [
        {
          "genre": "фэнтези"
        },
        {
          "genre": "боевик"
        }
      ],
      "rating": "6.0",
      "ratingVoteCount": 645,
      "posterUrl": "https://kinopoiskapiunofficial.tech/images/posters/kp/1236082.jpg",
      "posterUrlPreview": "https://kinopoiskapiunofficial.tech/images/posters/kp_small/1236082.jpg",
      "ratingChange": null
    }
  ]
}
