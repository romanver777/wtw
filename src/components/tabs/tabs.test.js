/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";

import Tabs from "./tabs";
import films from "../../mocks/films.json";
import { TABSNAME } from "../../helpers/const";
import Overview from "../overview/overview";
import Details from "../details/details";
import Reviews from "../reviews/reviews";

describe("Tabs component", () => {
  describe("with active tab - Overview", () => {
    const propsOverview = {
      film: films[1],
      reviews: films[1].reviews,
      staff: films[1].staff,
      initTab: TABSNAME[0],
    };
    const Trenderer = TestRenderer.create(<Tabs {...propsOverview} />);

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });
    it("renders active tab title", () => {
      const className = "movie-nav__item movie-nav__item--active";

      expect(Trenderer.root.findAllByType("li")[0].props.className).toEqual(
        className
      );
    });
    it("renders Overview props", () => {
      expect(Trenderer.root.findByType(Overview).props.film).toBe(
        propsOverview.film
      );
    });
  });

  describe("with active tab - Details", () => {
    const propsDetails = {
      film: films[1],
      reviews: films[1].reviews,
      staff: films[1].staff,
      initTab: TABSNAME[1],
    };
    const Trenderer = TestRenderer.create(<Tabs {...propsDetails} />);

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });
    it("renders active tab title", () => {
      const className = "movie-nav__item movie-nav__item--active";
      expect(Trenderer.root.findAllByType("li")[1].props.className).toEqual(
        className
      );
    });
    it("renders Details props", () => {
      expect(Trenderer.root.findByType(Details).props.staff).toBe(
        propsDetails.staff
      );
    });
  });

  describe("with active tab - Reviews", () => {
    const propsReviews = {
      film: films[2],
      reviews: films[2].reviews,
      staff: films[2].staff,
      initTab: TABSNAME[2],
    };
    const Trenderer = TestRenderer.create(<Tabs {...propsReviews} />);

    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });
    it("renders active tab title", () => {
      const className = "movie-nav__item movie-nav__item--active";
      expect(Trenderer.root.findAllByType("li")[2].props.className).toEqual(
        className
      );
    });
    it("renders Reviews props", () => {
      expect(Trenderer.root.findByType(Reviews).props.reviews).toBe(
        propsReviews.reviews
      );
    });
  });
});
