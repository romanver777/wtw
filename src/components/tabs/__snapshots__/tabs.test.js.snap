// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs component with active tab - Details renders correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Описание
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Информация
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Отзывы
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Режиссер
        </strong>
        <span
          className="movie-card__details-value"
        >
          Ли Дэниелс
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          В ролях
        </strong>
        <span
          className="movie-card__details-value"
        >
          Андра Дэй, 
          Гаррет Хедлунд, 
          Наташа Лионн, 
          Тайлер Джеймс Уильямс, 
          Треванте Роудс 
        </span>
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Длительность
        </strong>
        <span
          className="movie-card__details-value"
        />
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Жанр
        </strong>
        <span
          className="movie-card__details-value"
        >
          мультфильм, 
        </span>
        <span
          className="movie-card__details-value"
        >
          ужасы, 
        </span>
        <span
          className="movie-card__details-value"
        >
          фантастика
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Выход в прокат
        </strong>
        <span
          className="movie-card__details-value"
        >
          2021-...
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Tabs component with active tab - Overview renders correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Описание
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Информация
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Отзывы
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.1
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Превосходно
      </span>
      <span
        className="movie-rating__count"
      >
        1549
         голосов
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p />
    <p
      className="movie-card__director"
    >
      <strong>
        Режиссер: 
        Ли Дэниелс
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        В ролях: 
        Андра Дэй, 
        Гаррет Хедлунд, 
        Наташа Лионн, 
        Тайлер Джеймс Уильямс, 
        Треванте Роудс и другие
      </strong>
    </p>
  </div>
</div>
`;

exports[`Tabs component with active tab - Reviews renders correctly 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Описание
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Информация
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Отзывы
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__reviews movie-card__row"
  >
    <div
      className="movie-card__reviews-col"
    >
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Весьма рекомендую к просмотру фильм с очаровательной Андрой Дэй «Соединённые Штаты против Билли Холидей» (The United States vs. Billie Holiday, 2021) режиссера Ли Дэниелса. 
Не относитесь к нему, как к кино-шедевру, но и не сочтите его за грязную пропаганду BLM.
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              SlavaLim
            </cite>
            <time
              className="review__date"
              dateTime="2021-04-12T19:01:00"
            >
              12 апреля 2021 г., 19:01
            </time>
          </footer>
        </blockquote>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            В 1940-х правительство США начинает войну против наркотиков. Ведомые предубеждениями против цветного населения чиновники выбирают своей целью популярную певицу Билли Холидей.
Билли Холидей (англ. Billie Holiday), настоящее имя Элеанора Фейган (англ. Eleanora Fagan; 7 апреля 1915, Балтимор — 17 июля 1959,
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              DIMI25
            </cite>
            <time
              className="review__date"
              dateTime="2021-04-12T10:26:00"
            >
              12 апреля 2021 г., 10:26
            </time>
          </footer>
        </blockquote>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Биографическая драма Ли Дэниелса рассказывает о звезде 40—50-х годов, джазовой певице Билли Холидей. Основной акцент фильма делается на противостоянии Билли Холидей и ФБР. Бюро решило во что бы то ни стало запретить певице исполнять антирасистскую песню о судах Линча над чернокожими. И для этого агенты
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              an.varuhin
            </cite>
            <time
              className="review__date"
              dateTime="2021-03-22T12:29:00"
            >
              22 марта 2021 г., 12:29
            </time>
          </footer>
        </blockquote>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            За пару лет BLM набрало обороты столь быстрые и беспощадные, что пора уже всем сушить сухари. В Голливуде установились четкие правила: ни дня без кино на тему расовой дискриминации. Сладкие сны Спайка Ли реализовались. Не то чтобы это было категорически плохо, но уж больно попахивает конъектурой и социальным
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Ruslan Avdonin
            </cite>
            <time
              className="review__date"
              dateTime="2021-03-10T15:04:00"
            >
              10 марта 2021 г., 15:04
            </time>
          </footer>
        </blockquote>
      </div>
      <div
        className="review"
      >
        <blockquote
          className="review__quote"
        >
          <p
            className="review__text"
          >
            Одаренный режиссер Ли Дэниелс (автор «Сокровища», «Газетчика» и «Дворецкого») решил воплотить на экране жизнь дивы, настоящей иконы для американских чернокожих — Билли Холидей. Кинолента представляет собой достаточно прямой биографический фильм, раскрывающий последнее десятилетие жизни леди Дей: борьба
          </p>
          <footer
            className="review__details"
          >
            <cite
              className="review__author"
            >
              Egorche23
            </cite>
            <time
              className="review__date"
              dateTime="2021-03-03T19:48:00"
            >
              3 марта 2021 г., 19:48
            </time>
          </footer>
        </blockquote>
      </div>
    </div>
  </div>
</div>
`;
